lib/auth/auth.go

```diff
@@ -1671,6 +1671,21 @@ func (a *Server) SetAccessRequestState(ctx context.Context, params services.Acce
 	return trace.Wrap(err)
 }
 
+func (a *Server) GetAccessCapabilities(ctx context.Context, req services.AccessCapabilitiesRequest) (services.AccessCapabilities, error) {
+	var caps services.AccessCapabilities
+	if req.RequestableRoles {
+		v, err := services.NewRequestValidator(a, req.User)
+		if err != nil {
+			return services.AccessCapabilities{}, trace.Wrap(err)
+		}
+		caps.RequestableRoles, err = v.GetRequestableRoles()
+		if err != nil {
+			return services.AccessCapabilities{}, trace.Wrap(err)
```

Comment:
Once `services.AccessCapabilities` is an interface you can then just do `return nil, trace.Wrap(err)`. Our implementation will be consistent across the board then.